From d3cce0bbf7e327905965f0fae70b04af08bfb64b Mon Sep 17 00:00:00 2001
From: krebbel <krebbel@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 15 Apr 2011 12:43:40 +0000
Subject: [PATCH 124/200] 2011-04-15  Andreas Krebbel  <Andreas.Krebbel@de.ibm.com>

	* config/s390/s390.md (popcountdi2, popcountsi2, popcounthi2):
	Replace match_operand with match_dup for the third operand in
	these expanders.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@172491 138bc75d-0d04-0410-961f-82ee72b054a4

index 2d552fb..7639d2d 100644
--- a/gcc/config/s390/s390.md
+++ b/gcc/config/s390/s390.md
@@ -9330,7 +9330,7 @@
    (parallel [(set (match_dup 0) (plus:DI (match_dup 0) (match_dup 2)))
 	      (clobber (reg:CC CC_REGNUM))])
    ; sllg op2, op0, 16
-   (set (match_operand:DI 2 "register_operand" "")
+   (set (match_dup 2)
 	(ashift:DI (match_dup 0) (const_int 16)))
    ; agr op0, op2
    (parallel [(set (match_dup 0) (plus:DI (match_dup 0) (match_dup 2)))
@@ -9352,7 +9352,7 @@
 			      UNSPEC_POPCNT))
 	      (clobber (reg:CC CC_REGNUM))])
    ; sllk op2, op0, 16
-   (set (match_operand:SI 2 "register_operand" "")
+   (set (match_dup 2)
 	(ashift:SI (match_dup 0) (const_int 16)))
    ; ar op0, op2
    (parallel [(set (match_dup 0) (plus:SI (match_dup 0) (match_dup 2)))
@@ -9374,7 +9374,7 @@
 			      UNSPEC_POPCNT))
 	      (clobber (reg:CC CC_REGNUM))])
    ; sllk op2, op0, 8
-   (set (match_operand:SI 2 "register_operand" "")
+   (set (match_dup 2)
 	(ashift:SI (match_dup 0) (const_int 8)))
    ; ar op0, op2
    (parallel [(set (match_dup 0) (plus:SI (match_dup 0) (match_dup 2)))
-- 
1.7.0.4

